pkg/apis/apps/kfdef/v1beta1/testdata/kfctl_plugin_test.yaml

```diff
@@ -0,0 +1,8 @@
+apiVersion: kfdef.apps.kubeflow.org/v1beta1
+kind: KfDef
+spec:
+  plugins:
+    - name: fakeplugin
```

Comment:
@gabrielwen Are you still working on this test? Can we make this look like the spec in your comment?

```
apiVersion: kfdef.apps.kubeflow.org/v1beta1
kind: KfDef
spec:
  plugins:
    - kind: kfDefPlugin
      apiVersion: v1beta1
      metadata:
        name: gcp-plugin
      spec:
        kind: kfGcpPlugin
        apiVersion: v1alpha1
        metadata:
          name: gcp-plugin-spec
          namespace: kubeflow
        auth:
          iap:
            oAuthClientId: 147474701642-rn0ksbu4eikorfk8j0uvknubbke57ri1.apps.googleusercontent.com
            oAuthClientSecret:
              name: CLIENT_SECRET
        createPipelinePersistentStorage: true
```

Reply:
yes, I am.  of course I can change that.  before that... @kkasravi could you give me some pointers on how to generate deepcopy codes?  I keep getting errors when running `make deepcopy`

Reply:
How does the existing code work? I don't believe for GCPPluginSpec in v1alpha1 we are currently generating DeepCopy code.

Reply:
I need deepcopy code for `KfDef` in v1beta1, though.

Reply:
@gabrielwen why?

Reply:
because you need runtime.Object implementation to register the types:
https://github.com/kubeflow/kfctl/blob/48f8e149d7ddde344b64c6bc5b4508a2adbc5ae3/pkg/apis/apps/kfdef/v1alpha1/register.go#L50

anyway, I found the file.  it's in $GOPATH/src

Reply:
Great.

Reply:
test updated.